src/scheduler/fifo.c

```diff
@@ -564,9 +566,64 @@ schedule(int cmd, int sd, char *runjobid)
 				"reconfigure", "Scheduler is reconfiguring");
 			free_fairshare_head(conf.fairshare);
 			reset_global_resource_ptrs();
+
+			if (conf.prime_sort != NULL) {
```

Comment:
Put this code in init_config().  The point of that function is to initialize conf and make it ready for parsing the config file.  Checking for members that need to be freed is a fine thing to do there.  You could even move the freeing of free_fairshare_head() in there as well.

Reply:
Moved the code at the start of init_config.